python/pyspark/sql/udf.py

```diff
@@ -92,7 +92,7 @@ def __init__(self, func,
             raise TypeError(
                 "Invalid evalType: evalType should be an int but is {}".format(evalType))
 
-        self.func = func
+        self.func = fail_on_stopiteration(func)
```

Comment:
This is because of this from a very quick look. This looks __doc__. You should wrap it with `@functools.wraps`.